Kernel/Bus/USB/USBPipe.h

```diff
@@ -78,9 +71,45 @@ class Pipe {
     u16 m_max_packet_size { 0 };  // Max packet size for this pipe
     u8 m_poll_interval { 0 };     // Polling interval (in frames)
     bool m_data_toggle { false }; // Data toggle for stuffing bit
+};
 
-    Mutex m_dma_buffer_lock { "USB pipe mutex"sv };
+class ControlPipe : public Pipe {
```

Comment:
The commit description on the first commit could use some details beyond "refactor". maybe mention that there are now a series of subclasses for pipe?